consensus/core/src/storage/mod.rs

```diff
@@ -100,9 +103,20 @@ impl WriteBatch {
 
 /// Per-commit properties that can be derived and do not need to be part of the Commit struct.
 /// Only the latest version is needed for CommitInfo, but more versions are stored for
-/// debugging and potential recovery.
+/// debugging and potentially restoring from an earlier state.
 // TODO: version this struct.
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub(crate) struct CommitInfo {
-    pub(crate) last_committed_rounds: Vec<Round>,
+    pub(crate) committed_rounds: Vec<Round>,
+    pub(crate) timestamp_ms: BlockTimestampMs,
```

Comment:
When not writing `CommitInfo` for every round, the general flow during recovery would be
1. Read last `CommitInfo`, along with index and digest in the storage key.
2. Read `Commit` for the above index, and verify digest.
3. Read subsequent `Commit` and recalculate `CommitInfo` for them, including last committed rounds, commit timestamps and scores.
4. Get blocks for each `Commit`
5. Regenerate `CommittedSubDag`

Logic for deriving at least some fields of `CommitInfo` seems necessary. And getting blocks from store is necessary too to regenerating `CommittedSubDag`.

I know it is possible and cheap to keep timestamp in Commit. But I believe it will cause confusions since it is more similar to score and last committed rounds, and should be stored separately from `Commit`.